pkg/controller/release/release_controller.go

```diff
@@ -474,6 +474,23 @@ func (c *Controller) ensureReleaseState(relinfo *releaseInfo) (bool, []StrategyP
 		}
 	}
 	if succ != nil {
+		curTargetStep := relinfo.release.Spec.TargetStep
+		succTargetStep := succ.Spec.TargetStep
+
+		if curTargetStep < succTargetStep {
```

Comment:
Interesting question: what happens when the successor has more strategy steps than the current one (aka, a release that introduces a more complex strategy)?